### PR TITLE
use ->instance->defaultregion if ->instance->region doesn't exist for…

### PR DESCRIPTION
**Pull request by uzimike**
Potential fix for this plugin so it will play nice with blocks_multiblock.

If a block instance has no region, it will try and use its defaultregion field instead.

### CODE REVIEW

File: block.php [hunk 1]
```diff
@@ -27,7 +27,9 @@
 
 $action = optional_param('action', 'view', PARAM_ALPHANUMEXT);
 
-$PAGE->set_context(context_system::instance());
+require_login();
```
Comment (dmitriim) [inline block.php:30]: I think this not required as handled by a controller later in the code.  The same with` $PAGE->set_context(\context_system::instance());` Not obvious...

File: classes/block_manager.php [hunk 2]
```diff
@@ -361,14 +360,18 @@ public function edit_controls($block) {
         if (empty($blocktitle)) {
             $blocktitle = $block->arialabel;
         }
+        $blockregion = $block->instance->region;
```
Comment (dmitriim) [inline classes/block_manager.php:363]: are we 100% sure that $block->instance->region is always set? If not, maybe better this: 

`$blockregion = !empty($block->instance->region) ? $block->instance->region : $block->instance->defaultregion;`

        if (empty($blockregion)) {
            $blockregion = $block->instance->defaultregion;
        }

File: block.php (hunk 1 above)
Comment (dmitriim) [reply block.php:30]: ok. Looks like this is for CI to be happy. Should be ok to leave if doesn't trigger any errors/warnings.

File: classes/block_manager.php (hunk 2 above)
Comment (uzimike) [reply classes/block_manager.php]: It checks if $blockregion is empty just below, but this is tidier

**Pull request merged**